### DISCUSSION TITLE
Any Idea how would a Language Switcher Component be built using this?

### DISCUSSION THREAD

**Discussion by m1thrandir225** [Alexandre-Fernandez/astro-i18n#32] [Q&A]
I'm new to internationalization and I'm trying to build a language switcher with this library any idea of how it may be done? 

Thanks

**Comment by Alexandre-Fernandez** [comment]
```ts
astroI18n.langCodes
.filter(langCode => langCode !== astroI18n.langCode) // optional (excludes current language)
.map(langCode => l(Astro.url.pathname, {}, langCode)) // translate the url for every langCode
```
I didn't include a language switcher because the html structure may vary depending on your styles, but this is the idea.
Translate the current route for every available lang code. If your route have params you should pass them as the 2nd argument to the `l` function (you can access them with `Astro.params`).

**Reply by m1thrandir225** [reply]
Thanks, I did something similar just directly with 2 <a> tags, because I have only 2 languages, but in the future maybe an optional language switcher would be nice as an addition to this library. 

Thanks <3
